pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -6046,3 +6047,137 @@ func (r *ClickHouseReader) SearchTraces(ctx context.Context, params *model.Searc
 
 	return &searchSpansResult, nil
 }
+
+func (r *ClickHouseReader) GetCorrespondingNormalizedMetrics(ctx context.Context, orgID valuer.UUID, metricNames []string) (map[string]model.NormalizedMetricsMap, error) {
+	if len(metricNames) == 0 {
+		return make(map[string]model.NormalizedMetricsMap), nil
+	}
+
+	result := make(map[string]model.NormalizedMetricsMap)
+
+	sanitize := func(s string) string {
+		return constants.NormalizedMetricsMapRegex.ReplaceAllString(s, "")
+	}
+	stripQuantile := func(s string) string {
+		return constants.NormalizedMetricsMapQuantileRegex.ReplaceAllString(s, "")
+	}
+
+	metricsToRegex := func(name string) string {
+		tokens := strings.FieldsFunc(strings.ToLower(name), func(r rune) bool {
+			return !unicode.IsLetter(r) && !unicode.IsDigit(r)
+		})
+		if len(tokens) == 0 {
+			return "^$"
+		}
+		return "^" + strings.Join(tokens, "[^a-z0-9]*")
+	}
```

Comment:
Did you test the performance for this? The regex matching is 5x slower than directly using the metric and normalised metric name. For example, instead of building a regex for `process.cpu.time`, why can't we search on ['process.cpu.time', 'process_cpu_time']? We know the normalised metric name in the database, if any.